### DISCUSSION TITLE
使用了自动导入，那么size和zIndex要如何设置

### DISCUSSION THREAD

**Discussion by kailong321200875** [element-plus/element-plus#3829] [Q&A] [answered]
在vite中使用了ElementPlusResolver，会自动把组件按需导入进来，那么如何统一的去设置size和zIndex属性呢，是还要像官方文档写的那样

import ElementPlus from 'element-plus'
import App from './App.vue'

const app = createApp(App)
app.use(ElementPlus, { size: 'small', zIndex: 3000 })

还是说能在vite.config里面去直接设置，或者支持Config Provider组件去设置这两个属性吗

**Comment by jw-foss** [comment]
@kailong321200875 我觉得这是一个很好的点，我觉得这些东西都是需要通过 config provider 来做的，我会跟进一下。这么晚回复是在是非常抱歉。:bow:

**Reply by hooray** [reply]
赞同，全局性的设置放 config provider 更合理，现在 config provider 可配置的内容有点少

**Reply by msidolphin** [reply]
> 赞同，全局性的设置放 config provider 更合理，现在 config provider 可配置的内容有点少

我认为zIndex并不能放config provider配置，这个指不应该是一个props，size我们后续会支持的

**Comment by sxzz** [comment, marked as answer]
请参阅文档 https://element-plus.org/zh-CN/guide/quickstart.html#global-configuration

```js
app.config.globalProperties.$ELEMENT = {
  // options
}
```

**Reply by Alanz-design** [reply]
按需加载配置后按照 $ELEMENT 设置并没有起效欸，请问有其他实现方式吗
